src/App.tsx

```diff
@@ -5,7 +5,16 @@ import { FindMovie } from './components/FindMovie';
 import { Movie } from './types/Movie';
 
 export const App = () => {
-  const [movies] = useState<Movie[]>([]);
+  const [movies, setMovies] = useState<Movie[]>([]);
+
+  const addMovie = (newMovie: Movie) => {
+    if (movies.every(movie => (
+      movie.imdbId !== newMovie.imdbId))) {
+      setMovies((prev) => {
+        return [...prev, newMovie];
+      });
+    }
```

Comment:
do not use such complicated if's
better to find this movie before and then check